src/main/java/FlightGenerator.groovy

```diff
@@ -1,25 +1,112 @@
 import java.time.LocalDateTime
 
 class FlightGenerator {
+    static void main(String[] args) {
+        FlightGenerator fg = new FlightGenerator()
+        1.upto(10) { println(fg.generateFlightTo(Airports.SOFIA)) }
+    }
+
     final static int[] CAPACITIES = new int[]{80, 120, 150, 180, 240}
+    final static int FLIGHT_ID = 1
+    final Random rand = new Random()
+
+    Flight generateFlight(int id = getFlightID(),
+                          Airports from = getRandomAirport(rand),
+                          Airports destination = getRandomAirport(from, rand),
+                          LocalDateTime departure = getRandomDepartureDate(rand),
+                          LocalDateTime arrival = getRandomArrivalDate(departure, rand),
+                          int capacity = getRandomPlaneCapacity(rand),
+                          int passengerCount = getRandomPassengerCount(capacity, rand),
+                          boolean isRunningLate = getRandomIsLate(rand)) {
+        new Flight(id: id, from: from, destination: destination, departure: departure, arrival: arrival,
+                passengers: passengerCount, capacity: capacity, runningLate: isRunningLate)
+    }
+
+    Flight generateFlightTo(Airports destination) {
+        int id = getFlightID()
+        Airports from = getRandomAirport(destination, rand)
+        LocalDateTime departure = getRandomDepartureDate(rand)
+        LocalDateTime arrival = getRandomArrivalDate(departure, rand)
+        int capacity = getRandomPlaneCapacity(rand)
+        int passengerCount = getRandomPassengerCount(capacity, rand)
+        boolean isRunningLate = getRandomIsLate(rand)
+
+        generateFlight(id, from, destination, departure, arrival, capacity, passengerCount, isRunningLate)
+    }
+
+    Flight generateFlightFrom(Airports from) {
+        int id = getFlightID()
+        Airports destination = getRandomAirport(from, rand)
+        LocalDateTime departure = getRandomDepartureDate(rand)
+        LocalDateTime arrival = getRandomArrivalDate(departure, rand)
+        int capacity = getRandomPlaneCapacity(rand)
+        int passengerCount = getRandomPassengerCount(capacity, rand)
+        boolean isRunningLate = getRandomIsLate(rand)
+
+        generateFlight(id, from, destination, departure, arrival, capacity, passengerCount, isRunningLate)
+    }
```

Comment:
Even more clever would be to have a single method where you pass a Map/Flight as an input param. 
Whichever field is != null -> use that. Otherwise use the random value.
>Flight generateFlight(Map values) {
> Flight randomFlight = new Random values
> if(value.from) -> use it
> if(value.to) -> use it.
> ... 
> return randomFlight }